### DISCUSSION TITLE
Access member name in listing

### DISCUSSION THREAD

**Discussion by sar5430** [NationalSecurityAgency/ghidra#2772] [Q&A]
Hey ! I'm looking at the ghidra's source code trying to understand how I could implement a simple markup for structure member in the Listing view. To explain a bit more, I would like to see in the listing view the offset to a structure. For exemple, if `esi` contain a pointer to an type where the member at offset 0x4 is `field_4`, I would like to see `[esi + 4] => field_4` in the listing. It's already available in the decompiler view but I can't manage to find where it could be implemented in Listing. It is even implemented for global variables in listing, so I guess it should not be too hard to implement. Does anyone have an idea about that ?

**Comment by dragonmacher** [comment]
Ghidra will do this now in some situations.   That code is in `ghidra.program.model.listing.CodeUnitFormat`.

**Comment by sar5430** [comment]
Thanks for your answer. I did some tests and realized that it only works when the structure member have a defined address (ie when the structure is on the stack, or in `.data` section). Otherwise, the it will not appear, as the member of the structure does not have a "defined" address. I'm trying to understand how I could make it works, but looking at the code, it must not be that trivial. Have you any idea regarding that ?
